x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -194,6 +197,18 @@ private void innerExecute(Task task, EsqlQueryRequest request, ActionListener<Es
         );
     }
 
+    private EsqlExecutionInfo getOrCreateExecutionInfo(Task task, EsqlQueryRequest request) {
+        if (task instanceof EsqlQueryTask esqlQueryTask && esqlQueryTask.executionInfo() != null) {
```

Comment:
Can it really be anything but `EsqlQueryTask` at all?

Reply:
Yes, when it is a synchronous query request, the task is CancellableTask, not EsqlQueryTask. This method is called from innerExecute which both async and sync queries use, so this is the place where we decide whether to grab the executionInfo out of the task (async) or create it new for the first time (sync).